### ISSUE TITLE
hironx init pose may not be set factory initial value

### ISSUE THREAD

**Issue by 130s** [start-jsk/rtmros_hironx#107]
I don't know if this applies to all Hiro but for some newer robots, the pose `HIRONX.goInitial` sets is different from its factory initial pose.

For `NextageClient`, [it's adjusted](https://github.com/tork-a/rtmros_nextage/commit/1391e19961c80784e5ad510dcee1dd50aa708934#diff-74831e17eddab0218a7f6d7fbc024a30).

Unless there's a need I want the same value for `HIRONX` so that I can use the same code between Hiro and NXO.

I can't find a commit log where the current value was set ([log](https://code.google.com/p/rtm-ros-robotics/source/list?path=/trunk/rtmros_hironx/hironx_ros_bridge/src/hironx_ros_bridge/hironx_client.py&start=6889)).

**Comment by emijah** [comment]
It may have been there from the beginning.
The default value for GRX was 
   [-0.6,  0, -100,  15.2,  9.4,  3.2],
   [ 0.6,  0, -100, -15.2,  9.4, -3.2],
However, the new values should be the ones used to check joint alignment.

**Comment by k-okada** [comment]
what is `factory initial pose` ??? 
the pose which the robot is set after the calibration?

**Comment by 130s** [comment]
> what is `factory initial pose` ??? the pose which the robot is set after the calibration?

I meant `factory initial pose` as the one manufacturer uses for calibration. As I said, initial pose of `NextageClient` is already set with the manufacturer's value.

If this suggestion is acceptable, I'll apply the change. Then we need to notify the current users since this might break their application program.

**Comment by k-okada** [comment]
Let me clear the situation

What is the reason of changing initial pose to factory calibration pose?(What's the dis-advantage of `goInitial()` is different from `factory initial pose`?)

Does `factory initial pose` happen only after `checkEncoders()` ? what's will happens if we put `goInital` in the end of `checkEncoders()` function?

Basically,

```
                   [-0.6, 0, -140, 15.2, 9.4, 3.2],
                   [0.6, 0, -140, -15.2, 9.4, -3.2],
```

is better than 

```
                   [-0, 0, -130, 0, 0, 0],
                   [0, 0, -130, 0, 0, 0],
```

because, the angle of the end-effecter becomes 0. so that's seems better initial posture for users to start with, specially, they uses `setTargetPoseRelative()`

**Comment by 130s** [comment]
> What is the reason of changing initial pose to factory calibration pose?(What's the dis-advantage of goInitial() is different from factory initial pose?)

What I want is:
- (1st Priority) To have a common pose  `goInitial` in the class `HIRONX` and `NextageClient`. 
- To have factory init pose as a readily available option; For some hardware tests where manufacturer is involved (eg. acceptance test at their office, which most of all users basically have nothing to do with), they use factory initial pose.

> Does factory initial pose happen only after checkEncoders() ? what's will happens if we put goInital in the end of checkEncoders() function?

I doesn't matter when the factory init pose happens, I think.

> Basically,
> 
> ```
>                [-0.6, 0, -140, 15.2, 9.4, 3.2],
>                [0.6, 0, -140, -15.2, 9.4, -3.2],
> ```
> 
> is better than
> 
> ```
>                [-0, 0, -130, 0, 0, 0],
>                [0, 0, -130, 0, 0, 0],
> ```
> 
> because, the angle of the end-effecter becomes 0. so that's seems better initial posture for users to start with, specially, they uses setTargetPoseRelative()

I agree, and I'm sure @emijah does too.

So in conclusion, I guess I'll open a solution PReq that does:
- In `NextageClient`, set the same initial pose that `HIRONX` uses (ie. the one that sets the eef level up).
- In `HIRONX`, add an optional argument in `goInital` to set the factory initial pose as the initial pose.

**Comment by emijah** [comment]
Yes I agree too.
